### PR TITLE
Clean up `ProgressBar::class`.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌

### CODE REVIEW

File: docs/progressbar.md [hunk 1]
```diff
@@ -44,14 +44,16 @@ The code above generates the following HTML:
...
+
+All setters are immutable and return a new instance of the `Yiisoft\Yii\Bulma\ProgressBar` class with the specified value.
 
 Method | Description | Default
 -------|-------------|---------
-`id(string $value)` | Widget ID. | `''`
+`attributes(array $value)` | The HTML attributes. | `[]`
 `autoIdPrefix(string $value)` | Prefix to the automatically generated widget ID. | `w`
-`value(float $value)` | The progress value. Set to `0` to display a loading animation. | `0`
+`color(string $value)` | Bar color. Options available are: (`ProgressBar::COLOR_PRIMARY`, `ProgressBar::COLOR_LINK`, `ProgressBar::COLOR_INFO`, `ProgressBar::COLOR_SUCCESS`, `ProgressBar::COLOR_WARNING`, `ProgressBar::COLOR_DANGER`, `ProgressBar::COLOR_DARK`). | Default setting empty whitout color.
```
Comment (samdark) [inline docs/progressbar.md:55]:
```suggestion
`color(string $value)` | Bar color. Options available are: (`ProgressBar::COLOR_PRIMARY`, `ProgressBar::COLOR_LINK`, `ProgressBar::COLOR_INFO`, `ProgressBar::COLOR_SUCCESS`, `ProgressBar::COLOR_WARNING`, `ProgressBar::COLOR_DANGER`, `ProgressBar::COLOR_DARK`). | Default is no color.
```

File: docs/progressbar.md [hunk 2]
```diff
@@ -44,14 +44,16 @@ The code above generates the following HTML:
...
+`id(string $value)` | Widget ID. | `''`
 `maxValue(int $value)` | Maximum progress value. `0` means no maximum. | `100`
-`options(array $value)` | HTML attributes for the widget container tag. | `['class' => 'progress']`
-`size(string $value)` | Bar size. | `is-small`, `is-medium`, `is-large`
-`color(string $value)` | Bar color. | `is-primary`, `is-link`, `is-info`, `is-success`, `is-warning`, `is-danger`
+`size(string $value)` | Bar size. Options available are: (`ProgressBar::SIZE_SMALL`, `ProgressBar::SIZE_MEDIUM`, `ProgressBar::SIZE_LARGE`). | Default setting empty normal.
```
Comment (samdark) [inline docs/progressbar.md:58]:
```suggestion
`size(string $value)` | Bar size. Options available are: (`ProgressBar::SIZE_SMALL`, `ProgressBar::SIZE_MEDIUM`, `ProgressBar::SIZE_LARGE`). | Default setting is normal.
```

File: src/ProgressBar.php [hunk 3]
```diff
@@ -37,77 +39,119 @@ final class ProgressBar extends Widget
...
+        $new->autoIdPrefix = $value;
         return $new;
     }
 
     /**
-     * Returns a new instance with the specified value of the progress.
+     * Returns a new instance with the specified progress bar color.
      *
-     * @param float $value The value of the progress. Set `0` to display loading animation.
+     * @param string $value The progress bar color. Default setting color empty without any color.
```
Comment (samdark) [inline src/ProgressBar.php:91]:
```suggestion
     * @param string $value The progress bar color. By default there is no color.
```

File: src/ProgressBar.php [hunk 4]
```diff
@@ -37,77 +39,119 @@ final class ProgressBar extends Widget
...
+     * Possible values: Modal::COLOR_PRIMARY, Modal::COLOR_LINK, Modal::COLOR_INFO, Modal::COLOR_SUCCESS,
+     * Modal::COLOR_WARNING, Modal::COLOR_DANGER, Modal::COLOR_DARK.
      *
      * @return self
      */
-    public function value(float $value): self
+    public function color(string $value): self
     {
+        if (!in_array($value, self::COLOR_ALL, true)) {
+            $values = implode(' ', self::COLOR_ALL);
```
Comment (samdark) [inline src/ProgressBar.php:100]:
```suggestion
            $values = implode('", "', self::COLOR_ALL);
```

File: src/ProgressBar.php [hunk 5]
```diff
@@ -37,77 +39,119 @@ final class ProgressBar extends Widget
...
+        $new->attributes['max'] = $value;
 
         return $new;
     }
 
     /**
      * Returns a new instance with the specified progress bar size class.
      *
-     * @param string $value The progress bar size class.
+     * @param string $value The progress bar size class. Default setting empty normal.
```
Comment (samdark) [inline src/ProgressBar.php:146]:
```suggestion
     * @param string $value The progress bar size class. Default setting is "normal".
```

File: src/ProgressBar.php [hunk 6]
```diff
@@ -37,77 +39,119 @@ final class ProgressBar extends Widget
...
+     * Possible values: Modal::SIZE_SMALL, Modal::SIZE_MEDIUM, Model::SIZE_LARGE.
      *
      * @return self
      */
     public function size(string $value): self
     {
         if (!in_array($value, self::SIZE_ALL, true)) {
-            $values = implode('"', self::SIZE_ALL);
+            $values = implode(' ', self::SIZE_ALL);
```
Comment (samdark) [inline src/ProgressBar.php:154]:
```suggestion
            $values = implode('", "', self::SIZE_ALL);
```

File: src/ProgressBar.php [hunk 7]
```diff
@@ -118,56 +162,61 @@ public function size(string $value): self
...
      *
      * @return self
      */
-    public function color(string $value): self
+    public function value(?float $value): self
     {
-        if (!in_array($value, self::COLOR_ALL, true)) {
-            $values = implode('"', self::COLOR_ALL);
-            throw new InvalidArgumentException("Invalid color. Valid values are: \"$values\".");
+        if ($value < 0 || $value > 100) {
```
Comment (samdark) [inline src/ProgressBar.php:173]: Seems this condition edge case isn't covered by test.

**Review by samdark — CHANGES_REQUESTED** [review]

File: docs/progressbar.md (hunk 1 above)
Comment (terabytesoftw) [reply docs/progressbar.md:55]: done.

File: docs/progressbar.md (hunk 2 above)
Comment (terabytesoftw) [reply docs/progressbar.md:58]: done.

File: src/ProgressBar.php (hunk 3 above)
Comment (terabytesoftw) [reply src/ProgressBar.php:91]: done.

File: src/ProgressBar.php (hunk 4 above)
Comment (terabytesoftw) [reply src/ProgressBar.php:100]: done.

File: src/ProgressBar.php (hunk 5 above)
Comment (terabytesoftw) [reply src/ProgressBar.php:146]: done.

File: src/ProgressBar.php (hunk 6 above)
Comment (terabytesoftw) [reply src/ProgressBar.php:154]: done.

File: src/ProgressBar.php (hunk 7 above)
Comment (terabytesoftw) [reply src/ProgressBar.php:173]: https://github.com/terabytesoftw/yii-bulma/blob/959aa9e66629fe59094f330678aa74a38387a34f/tests/ProgressBarTest.php#L90

**Comment by samdark** [comment]
👍
